j5/boards/base.py

```diff
@@ -80,18 +77,20 @@ def __init__(self, board: Board, backend: Backend):
         self.board_class: Board = board
         self._backend: Backend = backend
         self._iterator_counter: int = 0
-        self.boards: List[Board] = []
+        self.boards: Dict[str, Board] = {}
 
         self.update_boards()
 
     def update_boards(self) -> None:
         """Update the boards in this group to see if new boards have been added."""
-        self.boards = self.board_class.discover(self._backend)
+        self.boards = {}  # type: Dict[str, Board]
+        for board in self.board_class.discover(self._backend):
+            self.boards[board.serial] = board
```

Comment:
`self.boards.update` would be more pythonic here imo.

```suggestion
            self.boards.update(board.serial: board)
```

Reply:
@trickeydan The syntax that this line has been changed to seems more convoluted than the original - what would you say makes it more Pythonic?